nikola/plugins/compile/markdown/__init__.py

```diff
@@ -43,40 +44,56 @@
 from nikola.utils import makedirs, req_missing, write_metadata
 
 
+class ThreadLocalMarkdown(threading.local):
+    """Convert to markdown using per-thread Markdown objects."""
+
+    def __init__(self, extensions):
+        self.markdown = Markdown(extensions=extensions, output_format="html5")
```

Comment:
I'm not sure whether this is the correct way to use `threading.local`. Every thread creates a copy of this initial value, but depending on how this copy is created (maybe even none at all, by just copying the reference to the `Markdown` object) the resulting object is not thread-safe.

It might be better to set `self.markdown` to `None`, and to check `self.markdown` in `convert()` and initialize it there when its current value is `None`.